src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -113,11 +142,42 @@ internal static IEnumerable<ConstructorInfo> GetConstructors(Type type, Type?[]
         /// </summary>
         internal static bool ParametersMatch(this ParameterInfo[] pinfos, Type?[] ptypes)
         {
-            if (pinfos.Length != ptypes.Length)
-                return false;
+            bool hasParamsArgument = pinfos.Length > 0 && pinfos[pinfos.Length -1].HasAttribute<ParamArrayAttribute>(false);
+
+            if (hasParamsArgument)
+            {
+                if (ptypes.Length < pinfos.Length -1)
+                    return false;
+            }
+            else
+            {
+                if (pinfos.Length != ptypes.Length)
+                    return false;
+            }
 
             for (int i = 0; i < pinfos.Length; i++)
             {
+                if (hasParamsArgument && i == pinfos.Length - 1)
+                {
+                    var elementType = pinfos[i].ParameterType.GetElementType();
+                    if (elementType is not null)
+                    {
+                        bool allParamArgsMatched = true;
+                        for (int j = i; j < ptypes.Length; j++)
+                        {
+                            if (!ptypes[j].CanImplicitlyConvertTo(elementType))
+                            {
+                                allParamArgsMatched = false;
+                                break;
+                            }
+                        }
+                        if (allParamArgsMatched)
+                        {
+                            continue;
+                        }
+                    }
+                }
+
                 if (!ptypes[i].CanImplicitlyConvertTo(pinfos[i].ParameterType))
```

Comment:
If you do this test first, and only if it fails AND we are dealing with the last `params` parameter, do the test above.

Reply:
Sorry I've tried switching it to do it this way round and it doesn't make the code easier because we need to special case it for the example of a params argument that ends up being passed no parameters at all

Reply:
After loading the code in VS, I concur